js/jquery.validation.js

```diff
@@ -0,0 +1,222 @@
+                                $form.validation('reposition');
+                            }
+                            // Reset the form
+                            $form.trigger('reset');
+                            
+                        } else if (response.message) {
+                            $.each(response.message, function(type, object) {
+                                $form.validation('showMessage', type, object.value);
+                                $form.validation('reposition');
+                            });
+                        }
+                        // Trigger validation completion event
+                        $form.trigger({
+                            type: 'validationComplete',
+                            response: response
+                        });
+                    },
+                    error: function(response, error, message) {
+                        var myWindow = window.open('', '_parent');
+                        myWindow.document.write(response.responseText);
+                    }
+                });
+            }
+            
+            $(this.element).on('submit.validate', validate);
+        },
+        
+        showValidationMessage: function($element, message) {
+            // Show the validation message with the message as the content for the given element.
+            var api = $element.qtip('api');
+            if ( api === undefined ) {
+                var qtipOptions = {
+                    content: message,
+                    position: {
+                        target: $element
+                    }
+                };
+                $.extend(true, qtipOptions, this.options.qtip);
+                $.each(this.options.hints, function(selector, hintOptions) {
+                    if ( $element.is(selector) ) {
+                        $.extend(true, qtipOptions, hintOptions);
+                    }
+                });
+                // initialise qtip
+                $element.qtip(qtipOptions);
+            } else {
+                // Update existing qtip and show
+                api.set('content.text', message);
+                api.reposition();
+                api.show();
+            }
+            return $element;
+        },
+        
+        hideValidationMessage: function($element) {
+            // Hide the validation message for the given element.
+            $element.qtip('hide');
+            return $element;
+        },
+        
+        showMessage: function(type, message) {
+            // Show the message of the type given with message as the content.
+            if (! this[type]) {
+                // Message area doesn't exist so create it.
+                var messageDiv = $('<div></div>').addClass(this.options.messages[type].classes)
+                var messageContent = $('<span></span>').text(message).addClass('message-content');
+                var messageClose = $('<span></span>').addClass('icon icon--cross');
```

Comment:
Not sure how we should deal with the close icon on messages. Should an icon font be included along with CSS for it?